pysph/examples/st_bouscasse.py

```diff
@@ -0,0 +1,190 @@
+"""Shallow water sloshing
+
+The case is as described in "Bouscasse, B., Antuono, M.,
+Colagrossi, A., & Lugni, C. (2013). Numerical and Experimental
+Investigation of Nonlinear Shallow Water Sloshing,
+International Journal of Nonlinear Sciences and Numerical Simulation,
+14(2), 123-138"
+
+DOI:10.1515/ijnsns-2012-0100
+
+This case is same as the Series 5 (Large amplitude sway motion with
+a small water depth) case mentioned in the above paper.
+"""
+
+from math import cos, pi, exp, sqrt, tanh, sin
+
+import os
+import numpy as np
+
+from pysph.base.utils import get_particle_array
+from pysph.base.kernels import CubicSpline
+from pysph.solver.solver import Solver
+from pysph.solver.application import Application
+
+from pysph.sph.integrator import EPECIntegrator
+
+from pysph.sph.equation import Equation
+from pysph.tools.geometry import get_2d_tank, get_2d_block
+from pysph.sph.wc.edac import EDACScheme, EDACStep
+
+
+L = 1
+h = 0.03 * L
+A = 2.333 * h
+
+u_max = 1.3
+c0 = 10.0 * u_max
+
+dx = h/6
+hdx = 1.2
+
+h0 = hdx * dx
+n_layers = 4
+
+tf = 33
+rho = 1000.0
+
+k = pi/L
+omega_r = sqrt(9.81*k*tanh(k*h))
+omega = omega_r*1.231
+
+
+class HorizontalExcitation(Equation):
+    def __init__(self, dest, sources):
+        super(HorizontalExcitation, self).__init__(dest, sources)
+
+    def initialize(self, d_idx, d_au, t, d_rho):
+        L = 1
+        h = 0.03 * L
+        A = 2.333 * h
+
+        k = pi/L
+        omega_r = sqrt(9.81*k*tanh(k*h))
+        omega = omega_r*1.231
```

Comment:
Pass the variables to `__init__` as equation parameters, don't declare them here. Avoid using single letter names to variables.